pkg/controller/secret/secret_string.go

```diff
@@ -148,3 +108,56 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+func (pg StringGenerator) determineKeysToRegenerate(instance *corev1.Secret, genKeys []string) (reconcile.Result, error) {
```

Comment:
I find the function name a bit misleading; "determine keys ..." suggests to me that the function actually only _determines_ which keys should be regenerated, but not _actually_ regenerating them. Maybe rename this...

```suggestion
func (pg StringGenerator) regenerateKeysWhereRequired(instance *corev1.Secret, genKeys []string) (reconcile.Result, error) {
```

❓